ui/src/app/nav.tsx

```diff
@@ -0,0 +1,120 @@
+            <div className="p-panel">
+                <div className="p-panel__header">
+                    <h4 className="p-panel__title">Add New CSR</h4>
+                    <div className="p-panel__controls">
+                        <button onClick={() => setIsOpen(false)} className="p-button--base u-no-margin--bottom has-icon"><i className="p-icon--close"></i></button>
+                    </div>
+                </div>
+                <div className="p-panel__content">
+                    <form className="p-form p-form--stacked">
+                        <div className="p-form__group row">
+                            <label htmlFor="textarea">Paste CSR in PEM format below</label>
+                            <textarea id="textarea" name="textarea" rows={10} placeholder="-----BEGIN CERTIFICATE REQUEST-----"></textarea>
+                        </div>
+                        <div className="p-form__group row">
+                            <button className="p-button--positive u-float-right" name="submit">Submit</button>
+                        </div>
+                    </form>
+                </div>
+            </div >
+        </aside >
+    )
+}
+
+export function SideBar({ sidebarVisible, setSidebarVisible }: { sidebarVisible: boolean, setSidebarVisible: Dispatch<SetStateAction<boolean>> }) {
+    const [activeTab, setActiveTab] = useState<string>("");
+
+    useEffect(() => {
+      if (typeof window !== 'undefined') {
+        setActiveTab(location.pathname.split('/')[1]);
+      }
+    }, []); 
+    return (
+        <header className={sidebarVisible ? "l-navigation" : "l-navigation is-collapsed"}>
+            <div className="l-navigation__drawer">
+                <div className="p-panel is-dark">
+                    <div className="p-panel__header is-sticky">
+                        <Logo />
+                        <div className="p-panel__controls u-hide--large">
+                            <button onClick={() => { setSidebarVisible(false) }} className="p-button--base is-dark has-icon u-no-margin u-hide--medium"><i className="is-light p-icon--close"></i></button>
+                            <button onClick={() => { setSidebarVisible(false) }} className="p-button--base is-dark has-icon u-no-margin u-hide--small"><i className="is-light p-icon--pin"></i></button>
+                        </div>
+                    </div>
+                    <div className="p-panel__content">
+                        <div className="p-side-navigation--icons" id="drawer-icons">
+                            <nav aria-label="Main">
+                                <ul className="p-side-navigation__list">
+                                    <li className="p-side-navigation__item">
+                                        <a className="p-side-navigation__link" href="/certificate_requests" aria-current={activeTab === "certificate_requests" ? "page" : "false"} >
```

Comment:
This is an interesting interaction, it looks like you need to be redirected to certificate_requests.html, which makes sense since this is a static HTML page. Fixed.

Reply:
for me it's the opposite
`http://localhost:3000/certificate_requests.html` leads me to a 404 and `http://localhost:3000/certificate_requests` shows actual content

Reply:
@saltiyazan I get the same thing as @gruyaume. are you running `npm run dev` within the `ui` subfolder? What you should do is run the following back to back:
`npm i --prefix ui`
`npm run build --prefix ui`
`go install ./...`
`gocert -config config.yaml`

Reply:
Ok now it does work fine but now we have to navigate to `.html` pages, ain't there a way to get directly to the specific endpoint?

Reply:
The way building works is nextjs produces a static file for each endpoint, and those static files are .html files. By embedding the `out` folder structure, we're basically serving a folder of files, including any types of files, just like how `python3 -m http.server` serves the cwd.

More reading [here](https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#deploying).


Obviously nginx is a no go, but I have some ideas about maybe possibly removing them in go directly:
* modify the files directly to remove the .html extension
* append .html to every request that gets sent to the frontend

It's probably best to tackle this in a different PR.

Reply:
Ok, I recommend looking at other Go project serving React UI's (ex. Vault, Parca), it's probably a solved problem. In either case, I'm fine with this being in a separate PR.

Reply:
ticket up at TLSENG-264